packages/back-end/src/services/datasource.ts

```diff
@@ -88,3 +91,47 @@ export async function testDataSourceConnection(
   const integration = getSourceIntegrationObject(datasource);
   await integration.testConnection();
 }
+
+export async function testQuery(
+  datasource: DataSourceInterface,
+  sql: string,
+  minExperimentLength: number,
+  startDate?: Date,
+  endDate?: Date,
+  experimentId?: string
+  // eslint-disable-next-line
+): Promise<TestQueryResult> {
+  const integration = getSourceIntegrationObject(datasource);
+
+  // The Mixpanel integration does not support test queries
+  if (!integration.testQuery) {
+    throw new Error("Unable to test query.");
+  }
+  return await integration.testQuery(
+    sql,
+    minExperimentLength,
+    startDate,
+    endDate,
+    experimentId
+  );
+}
+
```

Comment:
I've gone back and forth on this functionality. In the event that the test query doesn't return any rows, we can parse the query passed in, in order to get the list of columns, and then compare that to the required columns in order to warn users they passed in optional columns. 

But, I don't love the implementation, it feels like there could be an infinite amount of edge cases that might be hard to solve for, and I'm not sure the value is worth trying to support this.

Reply:
@jdorn We do have a check on the front-end for required columns - do you think it's worth it to check again here?

This check was originally checking for additional columns, rather than required columns.

Reply:
The front-end check is just a quick hack that doesn't actually see if the column is returned.  It just looks for the existence of strings in the query.  For example, the following would pass the front-end checks, but would fail when actually run:

```sql
-- timestamp, user_id, experiment_id, variation_id
SELECT 1 FROM table
```

Reply:
That being said, we don't need to run both queries as part of V1 of this feature.  Just the `SELECT *` version is fine for now.  We can expand it later with more sophisticated checks.